spec/integration/user_show_spec.rb

```diff
@@ -0,0 +1,21 @@
+require 'rails_helper'
+
+RSpec.describe 'Users', type: :system do
+  describe 'displays all users' do
+    it 'displays link to click' do
+      # display a page
+      visit users_path
+
+      # assert the displayed page
+      assert_text 'Tom'
+
+      # navigate to the show page
+      sleep(1)
+      click_on('Tom')
+
+      # assert the displayed page
+      sleep(1)
+      assert_text 'Bio'
+    end
+  end
+end
```

Comment:
- [ ] as part of the requirements, kindly add these missing tests for the user show: 

* Tests that asserts you can see the user's profile picture.
* Tests that asserts you can see the number of posts the user has written.
* Tests that asserts you can see the user's first 3 posts.
* Tests that asserts you can see a button that lets me view all of a user's posts.
* Tests that redirects to the post's show page when you click on a user's post.
* Tests that redirects to the user's post's index page when you click on a **see all posts**.